runtimes/native/src/runtime.c

```diff
@@ -83,6 +129,9 @@ void w4_runtimeBlitSub (const uint8_t* sprite, int x, int y, int width, int heig
     bool flipX = (flags & 2);
     bool flipY = (flags & 4);
     bool rotate = (flags & 8);
+    uint32_t bpp = (int)bpp2 + 1;
+    uint32_t nbits = mul_u32_with_overflow_check(mul_u32_with_overflow_check(width, height), bpp);
+    bounds_check(sprite, nbits / 8);
```

Comment:
I like the bounds check here, but do we necessarily need to do an overflow checked multiply? I think we can live with possible overflows, as long as it doesn't allow out of bounds access somehow?

Reply:
to implement bounds check that way, we need to make this module (runtime.c) know the implementation details of w4_framebufferBlit. or, implement bounds check in w4_framebufferBlit itself.

IMO, it's simpler to perform overflow checks here.

Reply:
Ah, I see. Passing huge values for width/height here that result in an overflow here would fool the bounds check and result in OOB in w4_framebufferBlit. Nice catch!